include/ESA.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef ESA_HPP
+#define ESA_HPP
+
+#include <limits>
+#include <map>
+#include "stls.hpp"
+
+// -----------------------------------------------------------------
+// Solver for the ESA scheme
+// -----------------------------------------------------------------
+
+class ESA : public Stls {
+
+protected:
+
+  // Function to perform the ESA scheme
+  void doESA();
+
+public:
+
+  // ESA constructor (inherited from Stls)
```

Comment:
Change the comment by removing the part in paranthesis. Since in the cpp file the constructor is not doing anything, just define the constructor in place by replacing the declaration with

ESA(const StlsInput& input) : Stls(input) { ; }